chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -815,10 +824,10 @@ impl Ledger {
         #[cfg(feature = "evm")]
         {
             // Set EVM environment values derived from block0 values
-            new_ledger.evm.environment.chain_id =
-                <[u8; 32]>::from(new_ledger.static_params.block0_initial_hash).into();
-            // TODO: set Origin, coinbase, and set block timestamp when
-            // they are defined. Using default values meanwhile.
+            new_ledger.evm.environment.chain_id = new_ledger.settings.evm_block_params.chain_id;
+            new_ledger.evm.environment.block_timestamp =
```

Comment:
I think it is better to move it inside `update_evm_block()` function

Reply:
This has been completely re-worked to go inside a method.